website/src/app/kb/deploy/gateways/readme.mdx

```diff
@@ -60,30 +60,30 @@ keep in mind when planning your deployment.
   correctly.
 </Alert>
 
-{/* ### Sizing recommendations */} {/* */}
-{/* Gateways, like the rest of Firezone's data plane stack, are written in Rust and */}
-{/* are thus Resource efficient by nature. */} {/* */}
-{/* A single Client connection to a single Gateway can typically reach speeds of **1 */}
-{/* Gbps** or more. This scales linearly with each Client connection up to the */}
-{/* number of CPU cores available to the Gateway. */} {/* */}
-{/* In our tests, we've found that even a single `f1.micro` instance on Google Cloud */}
-{/* Platform, equipped with a single shared vCPU core, can effortlessly handle up to */}
-{/* 1 Gbps of tunneled traffic. This performance is attributed to the efficiency of */}
-{/* the WireGuard® protocol and the Rust programming language. */} {/* */}
-{/* Use the table below as a rough guide for sizing your Gateway deployments. These */}
-{/* are general recommendations and may not be accurate for your specific use case. */}
-{/* If you're unsure, start with a smaller Gateway and scale up as needed. */}
-{/*  */}
-{/* | Gateway size | Users served   | CPU cores | Memory | Network link | */}
-{/* | ------------ | -------------- | --------- | ------ | ------------ | */}
-{/* | Micro        | 10 - 100       | 1         | 512 MB | 1 Gbps       | */}
-{/* | Small        | 100 - 250      | 2         | 1 GB   | 1 Gbps       | */}
-{/* | Medium       | 250 - 1,000    | 4         | 4 GB   | 10 Gbps      | */}
-{/* | Large        | 1,000 - 10,000 | 16        | 16 GB  | 25 Gbps      | */}
-{/*  */}
-{/* To go beyond the table above, you can deploy additional Gateways and use */}
-{/* Firezone's [automatic load balancing](#load-balancing) to distribute Client */}
-{/* connections across them. */}
+### Sizing recommendations
+
+The Gateway, like the rest of Firezone's data plane stack, is
+[written in Rust](https://github.com/firezone/firezone/tree/main/rust/gateway).
+As a result, it's extremely lightweight and computationally efficient by nature.
+
+Gateways are mostly performance-bound by the I/O rate at which they can process
+packets, so bare metal deployments or VMs with high I/O rates will make a big
+difference in both throughput and latency.
+
+Despite this, even tiny, shared vCPU VMs can perform acceptably well in many
+cases. An `f1-micro` instance with just 0.25 vCPU cores on Google Cloud, for
+example, can handle around 200 Mbps of sustained throughput.
+
+| Gateway size | Approx. users served | CPU cores | Memory | Recommended link speed |
+| ------------ | -------------------- | --------- | ------ | ---------------------- |
+| Micro        | 10 - 50              | 1         | 512 MB | 500 Mbps               |
+| Small        | 50 - 100             | 2         | 1 GB   | 1 Gbps                 |
+| Medium       | 100 - 500            | 4         | 2 GB   | 2.5 Gbps               |
```

Comment:
I don't think there will be much benefit beyond 2 cores right now, even 2 will probably see little difference.

The CPU speed will make a difference though at least for downloads and maybe we can say something about dedicated vs shared cpus

Reply:
Yeah I mentioned the shared thing above, and that I/O is king.

What we really want users to do is spin up more Gateways for more performance, because that's a unique differentiator for us